src/android/app/src/main/java/es/situm/plugin/PluginHelper.java

```diff
@@ -650,6 +652,50 @@ public void invalidateCache(CallbackContext callbackContext) {
         callbackContext.sendPluginResult(new PluginResult(Status.OK, "Cache invalidated"));
     }
 
+    public void updateNavigationState(final CordovaInterface cordova,
+        CordovaWebView webView,
+        JSONArray args,
+        final CallbackContext callbackContext) {
+            ExternalNavigation externalNavigation = null;
+            try {
+                JSONObject jsonObject = args.getJSONObject(0);
+                String type = jsonObject.getString("messageType");
+                ExternalNavigation.MessageType messageType;
+                switch (type) {
+                    case "NavigationStarted":
+                        messageType = ExternalNavigation.MessageType.NAVIGATION_STARTED;
+                        break;
+                  case "NavigationUpdated":
+                    messageType = ExternalNavigation.MessageType.NAVIGATION_UPDATED;
+                    break;
+                  case "DestinationReached":
+                    messageType = ExternalNavigation.MessageType.DESTINATION_REACHED;
+                    break;
+                  case "OutsideRoute":
+                    messageType = ExternalNavigation.MessageType.OUTSIDE_ROUTE;
+                    break;
+                  case "NavigationCancelled":
+                    messageType = ExternalNavigation.MessageType.NAVIGATION_CANCELLED;
+                    break;
+                    default:
+                      callbackContext.sendPluginResult(new PluginResult(Status.ERROR, "MessageType invalid"));
+                        return;
+                }
+                Map<String, Object> payload;
+                if (jsonObject.has("payload")) {
+                    payload = JsonUtils.toMap(jsonObject.getJSONObject("payload"));
+                  } else {
+                    payload = new HashMap<>();
+                  }
+                externalNavigation = new ExternalNavigation(messageType,payload);
+            } catch (JSONException e) {
+              callbackContext.sendPluginResult(new PluginResult(Status.ERROR, "Error parsing the external navigation json"));
```

Comment:
Pijada: pode axudar meter a excepcion ao final de `"Error parsing the external navigation json"` non si ?